src/main/java/org/jabref/gui/entryeditor/EntryEditor.java

```diff
@@ -463,11 +465,26 @@ private void fetchAndMerge(EntryBasedFetcher fetcher) {
 
     public void setFocusToField(Field field) {
         UiTaskExecutor.runInJavaFXThread(() -> {
+        Field actualField = field;
+        boolean fieldFound = false;
             for (Tab tab : tabbed.getTabs()) {
+                tabbed.getSelectionModel().select(tab);
                 if ((tab instanceof FieldsEditorTab fieldsEditorTab)
-                        && fieldsEditorTab.getShownFields().contains(field)) {
+                        && fieldsEditorTab.getShownFields().contains(actualField)) {
                     tabbed.getSelectionModel().select(tab);
-                    fieldsEditorTab.requestFocus(field);
+                    Platform.runLater(() -> {
+                        fieldsEditorTab.requestFocus(actualField);
+                    });
+                    // This line explicitly brings focus back to the main window containing the Entry Editor.
+                    getScene().getWindow().requestFocus();
```

Comment:
I would have wrapped that into `Platform.runlater`, too (and before the `runLater` above)